modin/pandas/groupby.py

```diff
@@ -1045,26 +1041,30 @@ def _wrap_aggregation(
         DataFrame or Series
             Returns the same type as `self._df`.
         """
+        if not isinstance(numeric_only, NumericOnly):
+            numeric_only = NumericOnly(numeric_only)
+
         agg_args = tuple() if agg_args is None else agg_args
         agg_kwargs = dict() if agg_kwargs is None else agg_kwargs
 
-        if numeric_only is None:
+        if numeric_only is NumericOnly.AUTO:
             # pandas behavior: if `numeric_only` wasn't explicitly specified then
             # the parameter is considered to be `False` if there are no numeric types
             # in the frame and `True` otherwise.
-            numeric_only = any(
-                is_numeric_dtype(dtype) for dtype in self._query_compiler.dtypes
+            numeric_only = NumericOnly(
+                any(is_numeric_dtype(dtype) for dtype in self._query_compiler.dtypes)
             )
 
-        if numeric_only and self.ndim == 2:
+        if numeric_only > NumericOnly.FALSE and self.ndim == 2:
```

Comment:
What?.. I don't think one should really be comparing enums on anything but (in)equality.

If you want to check on _classes_ of enum values, I'd suggest doing `IntFlag` instead

Reply:
I thought that the whole point of `IntEnum` is to work with them like an integer, e.g. `if day > DayEnum.TUESDAY: ...` or as in our case with 'numeric only' `if tolerance_level > ToleranceEnum.LEVEL2: ...`.

I've changed the [`IntEnum`](https://docs.python.org/3/library/enum.html#enum.IntEnum) to [`IntFlag`](https://docs.python.org/3/library/enum.html#enum.IntFlag) as you suggested, though don't see much difference between them for our use-case (even their docstrings from py-docs are almost identical).

Reply:
this is not what I meant... an `IntFlag` enum should be like
```python
class NumericOnly(IntFlag):
    AUTO = 0b000
    FALSE = 0b001
    TRUE = 0b010
    TRUE_EXCL_NUMERIC_CATEGORIES = 0b011
```
(I've stated the fields in bit notation for readability), and then you check stuff like
```python
if numeric_only & NumericOnly.TRUE:
    # do things
```

Reply:
Not everything which is technically allowed should be done. 🙃